src/combinatorics/simple_graph/inc_matrix.lean

```diff
@@ -64,125 +55,130 @@ def inc_matrix [has_zero R] [has_one R] : matrix α (sym2 α) R :=
 
 variables {R}
 
-@[simp]
-lemma inc_matrix_apply {a : α} {e : sym2 α} [has_zero R] [has_one R] :
+lemma inc_matrix_apply [has_zero R] [has_one R] {a : α} {e : sym2 α} :
   G.inc_matrix R a e = (G.incidence_set a).indicator 1 e := rfl
 
+/-- Entries of the incidence matrix can be computed given additional decidable instances. -/
+lemma inc_matrix_apply' [has_zero R] [has_one R] [decidable_eq α] [decidable_rel G.adj]
+  {a : α} {e : sym2 α} :
+  G.inc_matrix R a e = if e ∈ G.incidence_set a then 1 else 0 :=
+by convert rfl
+
 section mul_zero_one_class
 variables [mul_zero_one_class R] {a b : α} {e : sym2 α}
 
 lemma inc_matrix_apply_mul_inc_matrix_apply :
-  G.inc_matrix R a e * G.inc_matrix R b e =
-    (G.incidence_set a ∩ G.incidence_set b).indicator 1 e :=
-by rw [inc_matrix, ite_zero_mul_ite_zero_right _ _, mul_one]
--- by { convert set.indicator_mul_indicator _ _ _, }
+  G.inc_matrix R a e * G.inc_matrix R b e = (G.incidence_set a ∩ G.incidence_set b).indicator 1 e :=
+begin
+  simp only [inc_matrix, set.indicator_apply, ←ite_and_mul_zero,
+    pi.one_apply, mul_one, set.mem_inter_eq],
+  congr,
+end
 
 lemma inc_matrix_apply_mul_inc_matrix_apply_of_not_adj (hab : a ≠ b) (h : ¬ G.adj a b) :
   G.inc_matrix R a e * G.inc_matrix R b e = 0 :=
 begin
   rw [inc_matrix_apply_mul_inc_matrix_apply, set.indicator_of_not_mem],
-  exact λ h', h (G.adj_of_mem_incidence_set hab h'.1 h'.2),
+  rw [G.incidence_set_inter_incidence_set_of_not_adj h hab],
+  exact set.not_mem_empty e,
 end
 
 lemma inc_matrix_of_not_mem_incidence_set (h : e ∉ G.incidence_set a) :
   G.inc_matrix R a e = 0 :=
-if_neg sorry
+by rw [inc_matrix_apply, set.indicator_of_not_mem h]
 
 lemma inc_matrix_of_mem_incidence_set (h : e ∈ G.incidence_set a) : G.inc_matrix R a e = 1 :=
-if_pos sorry
+by rw [inc_matrix_apply, set.indicator_of_mem h, pi.one_apply]
 
 variables [nontrivial R]
 
 lemma inc_matrix_apply_eq_zero_iff : G.inc_matrix R a e = 0 ↔ e ∉ G.incidence_set a :=
-one_ne_zero.ite_eq_right_iff
+begin
+  simp only [inc_matrix_apply, set.indicator_apply_eq_zero, pi.one_apply, one_ne_zero],
+  exact iff.rfl,
+end
 
 lemma inc_matrix_apply_eq_one_iff : G.inc_matrix R a e = 1 ↔ e ∈ G.incidence_set a :=
-one_ne_zero.ite_eq_left_iff
```

Comment:
Why this regression?

Reply:
This would work too, but I didn't think to look deeper into it once I got everything working:
```
by { convert one_ne_zero.ite_eq_left_iff, assumption }
```
It's something to do with decidability typeclasses not being defeq.

Reply:
I'm being picky here because I wrote `ne.ite_eq_left_iff` specifically for this use 😛

Reply:
Ok, switched it back, wouldn't want to lose it.